game/src/sandbox/dashboards/commuter.rs

```diff
@@ -498,26 +502,71 @@ fn group_bldgs(
             id,
             bldgs: HashSet::new(),
             borders: hashset! { i.id },
-            shape: i.polygon.clone(),
+            shape: build_shape_for_border(i, BorderType::Incoming, &app.primary.map),
         });
     }
     for i in app.primary.map.all_outgoing_borders() {
-        if border_to_block.contains_key(&i.id) {
+        if let Some(incoming_border_id) = border_to_block.get(&i.id) {
+            let two_way_border = &mut blocks[*incoming_border_id];
+            two_way_border.shape = build_shape_for_border(i, BorderType::Both, &app.primary.map);
             continue;
         }
         let id = blocks.len();
         border_to_block.insert(i.id, id);
+
         blocks.push(Block {
             id,
             bldgs: HashSet::new(),
             borders: hashset! { i.id },
-            shape: i.polygon.clone(),
+            shape: build_shape_for_border(i, BorderType::Outgoing, &app.primary.map),
         });
     }
 
     (bldg_to_block, border_to_block, blocks)
 }
 
+enum BorderType {
+    Incoming,
+    Outgoing,
+    Both,
+}
+
+fn build_shape_for_border(
+    border: &map_model::Intersection,
+    border_type: BorderType,
+    map: &Map,
+) -> Polygon {
+    let start = border.polygon.center();
+
+    let road = match border_type {
```

Comment:
By definition, a border only has one road. You could just do `map.get_r(*border.roads.iter().next().unwrap())`. Although it's sort of gross to dig out an element guaranteed to exist in a `BTreeSet`.

Reply:
done!